modules/kernel/src/main/java/org/opencastproject/kernel/security/CustomPasswordEncoder.java

```diff
@@ -78,6 +78,8 @@ public boolean isPasswordValid(String encodedPassword, String rawPassword, Objec
       return encodedPassword.charAt(0) == '$' && BCrypt.checkpw(rawPassword, encodedPassword);
     } catch (IllegalArgumentException e) {
       logger.debug("bcrypt hash verification failed", e);
+    } catch (StringIndexOutOfBoundsException e) {
```

Comment:
I guess this is caused by `.charAt(0)`?
If yes, instead of catching the exception, my suggestion would be to use [`StringUtils#startsWith)[https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#startsWith-java.lang.CharSequence-java.lang.CharSequence-] instead which will gracefully handle this in the first place and also handle `null`.

Reply:
Seems my guess was correct. The cause is `charAt` and switching to `startsWith` should fix the underlying problem.

Reply:
Yep, my concern exactly.  I haven't had a chance to dig deeper into this, but this logic appears to check if there is an old md5'd password, and failing that checks for a bcrypt'd one.  If *neither* of these things are true, shouldn't this throw an exception?  A user without a valid form of password seems like a bug to me.

Reply:
No. You really don't want to throw an exception here if possible since this will cause Opencast to return an internal server error on login and may even disclose internal log-in credentials which is why I would also be careful with logging here. That's also part of why the logging for the `IllegalArgumentException` is just `debug`.